internal/comm/catchup_handler_test.go

```diff
@@ -247,3 +247,110 @@ func TestCatchupHandler_ServeHTTP_Blocks(t *testing.T) {
 		require.Equal(t, uint64(6), bNum)
 	})
 }
+
+func TestCatchupHandler_ServeHTTP_LargeResponse(t *testing.T) {
+	lg, err := logger.New(&logger.Config{
+		Level:         "debug",
+		OutputPath:    []string{"stdout"},
+		ErrOutputPath: []string{"stderr"},
+		Encoding:      "console",
+	})
+	require.NoError(t, err)
+
+	b1Size := 0
+	b5Size := 0
+
+	ledger1 := &memLedger{}
+	for n := uint64(1); n < 11; n++ {
+		block := &types.Block{
+			Header: &types.BlockHeader{
+				BaseHeader:           &types.BlockHeaderBase{Number: n},
+				TxMerkelTreeRootHash: make([]byte, 1024), //just for size
+			},
+		}
+		ledger1.Append(block)
+
+		if n==1 {
+			b1Size = len(httputils.MarshalOrPanic(block))
+		}
+		if n<6 {
+			b5Size += len(httputils.MarshalOrPanic(block))
+		}
+	}
+
+	t.Run("too many blocks in request", func(t *testing.T) {
+		h := comm.NewCatchupHandler(lg, ledger1, b5Size+1) // 5 blocks in response
```

Comment:
Is there any reason to do `b5Size+1` instead of `b5Size`? I assume none and both should work.

Reply:
Done